include/seqan/platform.h

```diff
@@ -398,6 +398,17 @@ typedef int8_t __int8;     // nolint
 // backwards compatibility
 #define SEQAN_UNUSED_TYPEDEF SEQAN_UNUSED
 
+// eliminate fallthrough warnings
+#if defined __has_cpp_attribute && __has_cpp_attribute(fallthrough)
```

Comment:
You need to make this in two consecutive if statements or it will fail on some of our platforms (see travis). It is probably easiest to do it like this:
```c++
#define SEQAN_FALLTHROUGH
#if defined(__has_cpp_attribute)
  #if __has_cpp_attribute(fallthrough)
    #undef SEQAN_FALLTHROUGH
    #if __cplusplus < 201500
      #define SEQAN_FALLTHROUGH [[gnu::fallthrough]];
    #else
      #define SEQAN_FALLTHROUGH [[fallthrough]];
    #endif
  #endif
#endif
```